zerver/lib/narrow.py

```diff
@@ -366,7 +366,11 @@ def by_is(self, query: Select, operand: str, maybe_negate: ConditionTransform) -
         elif operand == "mentioned":
             cond1 = column("flags", Integer).op("&")(UserMessage.flags.mentioned.mask) != 0
             cond2 = column("flags", Integer).op("&")(UserMessage.flags.wildcard_mentioned.mask) != 0
-            cond = or_(cond1, cond2)
+            cond3 = (
+                column("flags", Integer).op("&")(UserMessage.flags.topic_wildcard_mentioned.mask)
+                != 0
+            )
+            cond = or_(cond1, cond2, cond3)
             return query.where(maybe_negate(cond))
```

Comment:
I guess because this is SQLAlchemy, we can't use a common `models.py` function -- have you done an `EXPLAIN ANALYZE` of the SQL generated to confirm the `has:mention` narrow in fact uses this index?

Reply:
OK, that query plan looks good. Though we may need to recheck this if we end up changing the index structure.

Reply:
https://github.com/zulip/zulip/pull/27278#discussion_r1365821523